### ISSUE TITLE
Unexpected hydration attribute mismatch, with a ref bool attribute 

### ISSUE THREAD

**Issue by Jannchie** [nuxt/nuxt#25154]
### Environment

"nuxt": "^3.9.0"

### Reproduction

https://codesandbox.io/p/devbox/intelligent-kepler-lv49z8?file=%2Fapp.vue%3A11%2C36

It doesn't happen on stackblitz, but it happens locally and on codesandbox.

```vue
<script setup lang="ts">
import { NuxtLink } from '#components';

const disabled = ref(false);
</script>
<template>
  <div>
    <component :is="NuxtLink" :disabled="disabled" href="https://www.google.com">
      test
    </component>
  </div>
</template>
```

### Describe the bug

It looks like there's a problem with server-side rendering, it shouldn't render `disabled=""`

### Logs

```shell-script
[Vue warn]: Hydration attribute mismatch on <a href=​"https:​/​/​www.google.com" rel=​"noopener noreferrer">​ t ​</a>​ 
  - rendered on server: disabled=""
  - expected on client: (not rendered)
  Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
  You should fix the source of the mismatch. 
  at <NuxtLink disabled=false href="https://www.google.com" > 
  at <App key=3 > 
  at <NuxtRoot>
```

**Comment by manniL** [comment]
This is a Vue SSR issue.

**Comment by mrchoke** [comment]
> This is a Vue SSR issue and fixed in Vue `3.4.8`

I still found this in follower env

```console
+ vue 3.4.13
+ nuxt 3.9.1
+ nuxt-quasar-ui 2.0.7
+ quasar 2.14.2
```
```js
 [Vue warn]: Hydration attribute mismatch on <input class=​"q-field__native q-placeholder" style tabindex=​"0" aria-label=​"Full Name" autocompete=​"full-name" id=​"f_79587d3f-3ead-4df2-a28d-ecec7a611d65" type=​"text" value=​"John Doe">​ 
  - rendered on server: id="f_951ce5ac-d451-4d3c-830c-3caf7e4fdada"
  - expected on client: id="f_fcb56126-ded2-47fe-9f38-e1018dfa00e5"
  Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
  You should fix the source of the mismatch. 
  at <QInput modelValue="John Doe" onUpdate:modelValue=fn error-message=undefined  ... > 
  at <TextInput name="full_name" required="" label="Full Name"  ... > 
  at <QForm autocomplete="on" > 
  at <QPage padding="" > 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <Anonymous key="/" vnode= {__v_isVNode: true, __v_skip: true, type: {…}, props: {…}, key: null, …} route= {fullPath: '/', hash: '', query: {…}, name: 'index', path: '/', …}  ... > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <QPageContainer> 
  at <QLayout view="lHh LPr lFf" container="" style= {minHeight: '10px', minWidth: '280px', maxWidth: '1280px', margin: '0px auto'}  ... > 
  at <Default ref=Ref< undefined > > 
  at <LayoutLoader key="default" layoutProps= {ref: RefImpl} name="default" > 
  at <NuxtLayoutProvider layoutProps= {ref: RefImpl} key="default" name="default"  ... > 
  at <NuxtLayout> 
  at <App key=3 > 
  at <NuxtRoot>
```

**Comment by manniL** [comment]
That's a different story as it is id-related. shouldn't break anything though and is due to the lack of a "useId" function.
